File: server-dotnet/src/KiwiCart.Core/Entities/StoreToken.cs
namespace KiwiCart.Core.Entities;

public class StoreToken
{
    public int Id { get; set; }
    public string StoreBrand { get; set; } = string.Empty;
    public string Token { get; set; } = string.Empty;
    public DateTime IssuedAt { get; set; }
    public DateTime ExpiresAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public bool IsExpired => DateTime.UtcNow >= ExpiresAt;
    public bool IsNearExpiry => DateTime.UtcNow >= ExpiresAt.AddMinutes(-5);
}

File: server-java/src/main/java/nz/co/kiwicart/entity/StoreToken.java
package nz.co.kiwicart.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "store_tokens")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StoreToken {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "store_brand", nullable = false, unique = true)
    private String storeBrand;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String token;

    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    @PreUpdate
    protected void onSave() {
        updatedAt = LocalDateTime.now();
    }
}
